### PR TITLE
feat: add Azure FunctionApp Module

### PR DESCRIPTION
**Pull request by polatengin**
This PR is for adding Azure Function App module to the Terratest.

With this module, it can be possible to test Azure Function App resources on Terraform scripts.

### CODE REVIEW

**Comment by HadwaAbdelhalem** [comment]
Hi @yorinasub17, PR is updated and ready for your review. Here is the related [CI](https://github.com/polatengin/terratest/runs/3081781881?check_suite_focus=true). the failure in it is unrelated to the change, but it is due to Azure [failure](https://github.com/polatengin/terratest/runs/3081781881?check_suite_focus=true#step:10:2072)  to create cosmos DB due to unavailability

File: modules/azure/appService.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/web/mgmt/2019-08-01/web"
+	"github.com/stretchr/testify/require"
+)
+
+// AppExistsE indicates whether the specified application exists.
```
Comment (yorinasub17) [inline modules/azure/appService.go:11]:
```suggestion
// AppExists indicates whether the specified application exists.
```

**Review by yorinasub17 — COMMENTED** [review]
LGTM! I just found one typo in the docs. I'll kick off a regression build now to make sure our precommit hooks run.

File: modules/azure/appService.go (hunk 1 above)
Comment (HadwaAbdelhalem) [reply modules/azure/appService.go]: Fixed

**Review by yorinasub17 — APPROVED** [review]
LGTM and sanity check build passed, so we can merge this in!

**Pull request merged**